src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala

```diff
@@ -85,7 +87,7 @@ object TezosTypes {
     chain_id: Option[String],
     hash: BlockHash,
     header: BlockHeader,
-    metadata: BlockHeaderMetadata
+    metadata: Either[BlockHeaderMetadata, GenesisMetadata.type]
```

Comment:
As we understand it, `Either` is by convention used to capture and handle error messages using `Left`. Can we instead use a sum type here instead?

Reply:
`Either` can generally be used to represent a common "Sum type" with 2 alternatives.
It's commonly used to represent the sum of possible results of an operation when you can have a successful or failed outcome. But that's not a rule, it's a convention.

An alternative encoding of sum types is by shallow sealed trait hierachies (ADTs).
I will try to swap definition using the latter and see if the result is more palatable, in fact